core/src/main/c/share/vec_int_key_agg.cpp

```diff
@@ -286,7 +289,62 @@ static jboolean kIntMinLong(to_int_fn to_int, jlong pRosti, jlong pKeys, jlong p
     return JNI_TRUE;
 }
 
-static jboolean kIntMinDouble(to_int_fn to_int, jlong pRosti, jlong pKeys, jlong pDouble, jlong count, jint valueOffset) {
+static jboolean kIntMinShort(to_int_fn to_int, jlong pRosti, jlong pKeys, jlong pLong, jlong count, jint valueOffset) {
+    auto map = reinterpret_cast<rosti_t *>(pRosti);
+    const auto *pi = reinterpret_cast<jshort *>(pLong);
+    const auto value_offset = map->value_offsets_[valueOffset];
+    for (int i = 0; i < count; i++) {
+        MM_PREFETCH_T0(pi + i + 8);
+        const int32_t key = to_int(pKeys, i);
+        const jshort val = pi[i];
+        auto res = find(map, key);
+        auto pKey = map->slots_ + res.first;
+        auto pVal = pKey + value_offset;
+        if (PREDICT_FALSE(res.second)) {
+            if (PREDICT_FALSE(res.first == UL_MAX)) {
+                return JNI_FALSE;
+            }
+            *reinterpret_cast<int32_t *>(pKey) = key;
+            *reinterpret_cast<jlong *>(pVal) = val;
+        } else {
+            const jlong old = *reinterpret_cast<jlong *>(pVal);
+            if (old != L_MIN) {
+                *reinterpret_cast<jlong *>(pVal) = MIN(val, (jshort) old);
```

Comment:
I think it should be
```suggestion
             if (val != I_MIN) {
                 *reinterpret_cast<jlong *>(pVal) = MIN(val, (jshort) old);
             }
```